google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/GetProductBiddingCategoryConstant.java

```diff
@@ -0,0 +1,193 @@
+    if (!params.parseArguments(args)) {
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("ENTER_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetProductBiddingCategoryConstant().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   */
+  private static void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    // Creates the query.
+    String query =
+        "SELECT "
+            + "product_bidding_category_constant.localized_name, "
+            + "product_bidding_category_constant.product_bidding_category_constant_parent "
+            + "FROM "
+            + "product_bidding_category_constant "
+            + "WHERE "
+            + "product_bidding_category_constant.country_code IN ('US')";
+
+    // Creates the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setPageSize(PAGE_SIZE)
+            .setCustomerId(Long.toString(customerId))
+            .setQuery(query)
+            .build();
+
+    // Creates the Google Ads Service Client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+
+      // Creates a list of top level category nodes.
+      List<CategoryNode> rootCategories = new ArrayList<>();
+      // Creates a map of category ID to category node for all categories found in the results.
+      HashMap<String, CategoryNode> biddingCategories = new HashMap<>();
```

Comment:
This Map is a convenience lookup to enable fast retrieval of existing nodes. I would state that for clarity in the comments.